hayeon/programmers-1/행렬의-덧셈.js

```diff
@@ -0,0 +1,48 @@
+function solution(arr1, arr2) {
+  let result = [[]];
+  for (let i = 0; i < arr1.length; i++) {
+    result[i] = [];
+    for (let j = 0; j < arr1[i].length; j++) {
+      result[i].push(arr1[i][j] + arr2[i][j]);
+    }
+  }
+  return result;
+}
+
+/*
+처음 풀이
+function solution(arr1, arr2) {
+  let result = [];
+  let num = [];
+
+  for (let i = 0; i < arr1.length; i++) {
+    result.push(num);
+
+    for (let j = 0; j < arr1[i].length; j++) {
+      num.push(arr1[i][j] + arr2[i][j]);
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(arr1, arr2) {
  const result = [];
  let num = [];

  for (let i = 0; i < arr1.length; i++) {
    num = [];

    for (let j = 0; j < arr1[i].length; j++) {
      num.push(arr1[i][j] + arr2[i][j]);
    }
    result.push(num);
  }
  return result;
}
```
- 해당 풀이는 for문 첫번째 바로 result.push(num)을 하기보다 num을 빈 배열로 초기화하고, 2번째 for문에서 이 num이라는 배열에다 값을 넣어서 2번째 for문이 종료되면 result에 push하는 방향이 맞습니다 👍